src/features/areas/components/AreaOverlay/index.tsx

```diff
@@ -77,8 +78,11 @@ const AreaOverlay: FC<Props> = ({
   );
 
   useEffect(() => {
-    setTitle(area.title);
-    setDescription(area.description);
+    if (area.id !== areaId.current) {
```

Comment:
I think that instead of this, you could have just put `area.id` as the dependency in the dependency array instead of `area`, i.e.:

```ts
useEffect(() => {
  setTitle(area.title);
  setDescription(area.description);
}, [area.id]);
```